refinery/core/api.py

```diff
@@ -678,24 +671,34 @@ def filter_by_group(self, request, obj_list):
         return obj_list
 
     def obj_get(self, bundle, **kwargs):
-        dataset = SharableResourceAPIInterface.obj_get(self, bundle, **kwargs)
-        return self.handle_incomplete_dataset(dataset)
+        try:
+            dataset = DataSet.objects.get(uuid=kwargs["uuid"])
+        except (DataSet.DoesNotExist, DataSet.MultipleObjectsReturned) as e:
+            logger.error("Couldn't properly fetch DataSet with UUID: %s %s",
+                         kwargs["uuid"], e)
+            raise NotFound(e)
+
+        if not dataset.is_valid():
+            raise NotFound(
+                "DataSet with UUID: {} is invalid, and most likely is "
+                "still being created".format(dataset.uuid)
+            )
+        return SharableResourceAPIInterface.obj_get(self, bundle, **kwargs)
```

Comment:
Making this proposed change yields: 
`{AttributeError}'super' object has no attribute 'obj_get'`